doc/man7/provider.pod

```diff
@@ -260,25 +260,31 @@ algorithm identifier to the appropriate fetching function.
 
 The default provider is built in as part of the F<libcrypto> library.
 Should it be needed (if other providers are loaded and offer
-implementations of the same algorithms), the property "default=yes"
-can be used as a search criterion for these implementations.
+implementations of the same algorithms), the property "provider=default"
+can be used as a search criterion for these implementations. Some
+non-cryptographic algorithms (such as serializers for loading keys and
+parameters from files) are not FIPS algorithm implementations in themselves but
+support algorithms from the FIPS provider and are allowed for use in "FIPS
+mode". The property "fips=yes" can be used to select such algorithms.
 
 =head2 FIPS provider
 
 The FIPS provider is a dynamically loadable module, and must therefore
 be loaded explicitly, either in code or through OpenSSL configuration
 (see L<config(5)>).
 Should it be needed (if other providers are loaded and offer
-implementations of the same algorithms), the property "fips=yes" can
-be used as a search criterion for these implementations.
+implementations of the same algorithms), the property "provider=fips" can
+be used as a search criterion for these implementations. All algorithm
+implementations in the FIPS provider can also be selected with the property
```

Comment:
I'm still expecting e.g. MD5 to be in the FIPS provider (for TLS).  It can't be selected using "fips=yes".  It could be selected using "provider=fips" and then we're back to the beginning of this mess.

I don't think that this should prevent this PR moving forward.

Reply:
> I'm still expecting e.g. MD5 to be in the FIPS provider (for TLS)

I don't see how.  Looking at the IGhttps://csrc.nist.gov/csrc/media/projects/cryptographic-module-validation-program/documents/fips140-2/fips1402ig.pdf page 15, it says that if SHA1 is available than MD5 must not be used, for example.

Reply:
Several TLS versions require both SHA1 and MD5 to be present.

Reply:
These fall into the category of "Non-FIPS Approved But Allowed Cryptographic Functions" See SP 800-135

Reply:
Is TLS-1.0, 1.1 still something to worry about for FIPS? I would say that with the dependence of these protocol versions on SHA1 and prevalence of TLS-1.2 they do not have to be supported in FIPS mode anymore.